org.uqbar.project.wollok/src/org/uqbar/project/wollok/scoping/cache/MapBasedWollokGlobalScopeCache.xtend

```diff
@@ -14,36 +13,60 @@ class MapBasedWollokGlobalScopeCache implements WollokGlobalScopeCache {
 	val cache = new LRUMap(20)
 	
 	override get(URI uri, Set<String> imports, Function0<Iterable<IEObjectDescription>> ifAbsentBlock) {
-		val uriString = uri.toString
-		var cacheContent = cache.get(uriString) as MapBasedCacheContent
-		
-		if(cacheContent === null || cacheContent.mismatch(imports)){
-			cacheContent = new MapBasedCacheContent(uri, imports, ifAbsentBlock.apply)
-			cache.put(uriString, cacheContent)
+		synchronized (this) {
+			val uriString = uri.toString
+			var cacheContent = cache.get(uriString) as MapBasedCacheContent
+			if (cacheContent === null || cacheContent.mismatch(imports)) {
+				cacheContent = new MapBasedCacheContent(uri, imports, ifAbsentBlock.apply)
+				cache.put(uriString, cacheContent)
+			}
+			cacheContent.result
+		}
+	}
+	
+	override invalidateDependencies(URI uri) {
+		synchronized (this) {
+			val Set<String> uris = cache.keySet
+			
+			uris.clone.filter [ String uriDependency | 
+				(cache.get(uriDependency) as MapBasedCacheContent).hasDependencyWith(uri)
+			].forEach [
+				cache.remove(it)
+			]
+			cache.remove(uri.toString)
 		}
-		
-		cacheContent.result
 	}
 	
 	override clearCache() {
```

Comment:
this is the same as  `override synchronized clearCache()`

Reply:
Yes, I will change it.

Reply:
I did that way because I was not sure the behavior of a method when you are overriding an interface/abstract class method.